Expand autodiff view exclusion to tensor list operators

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66978
* #66977


See: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/graph_executor.cpp#L410

Previously other view operators were handled, but not tensor lists views. Additionally the base PR these operators aliasing were not correctly specified.

Differential Revision: [D31935572](https://our.internmc.facebook.com/intern/diff/D31935572)

Commit:
Expand autodiff view exclusion to tensor list operators

[ghstack-poisoned]

Commit:
Update on "Expand autodiff view exclusion to tensor list operators"




[ghstack-poisoned]

dzhulgakov reviewed (commented):
Not sure how to review the actual aliasing logic, but looks plausible :)

Commit:
Update on "Expand autodiff view exclusion to tensor list operators"



See: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/graph_executor.cpp#L410

Previously other view operators were handled, but not tensor lists views. Additionally the base PR these operators aliasing were not correctly specified.

[ghstack-poisoned]

Commit:
Update on "Expand autodiff view exclusion to tensor list operators"



See: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/graph_executor.cpp#L410

Previously other view operators were handled, but not tensor lists views. Additionally the base PR these operators aliasing were not correctly specified.

[ghstack-poisoned]

Commit:
Update on "Expand autodiff view exclusion to tensor list operators"



See: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/graph_executor.cpp#L410

Previously other view operators were handled, but not tensor lists views. Additionally the base PR these operators aliasing were not correctly specified.

Differential Revision: [D31935572](https://our.internmc.facebook.com/intern/diff/D31935572)

[ghstack-poisoned]
